code/datums/components/slippery.dm

```diff
@@ -48,6 +48,10 @@
 	Additionally calls the parent's `after_slip()` proc on the `victim`.
 */
 /datum/component/slippery/proc/Slip(datum/source, mob/living/carbon/human/victim)
-	if(istype(victim) && !victim.flying && prob(slip_chance) && victim.slip(description, knockdown, slip_tiles, walking_is_safe, slip_always, slip_verb))
+	if(istype(victim) && !victim.flying)
 		var/atom/movable/owner = parent
-		owner.after_slip(victim)
+		if(owner && iscloset(owner.loc))
+			return
+		if(prob(slip_chance) && victim.slip(description, knockdown, slip_tiles, walking_is_safe, slip_always, slip_verb))
+			if(owner)
+				owner.after_slip(victim)
```

Comment:
I would prefer if we didn't have `if(owner)` safety checks. If a component's `parent` doesn't exist, something has gone *very* wrong and it should runtime.

Reply:
Presumably the component system isn't limited to being used on `atom/movable`, though I can understand why in practice slippery would only be on those.

Anyhow, I've removed the safety check...